src/js/common/containers/DowntimeNotification.jsx

```diff
@@ -0,0 +1,248 @@
+
+    if (!downtime) return serviceStatus.ok;
+    if (current.isBetween(downtime.startTime, downtime.endTime, inclusive)) return serviceStatus.down;
+    if (warning.isBetween(downtime.startTime, downtime.endTime, inclusive)) return serviceStatus.downtimeApproaching;
+
+    return serviceStatus.ok;
+  }
+
+  getDowntimeWindow(downtimes) {
+    return downtimes.reduce((result, downtime) => {
+      if (!downtime.startTime || !downtime.endTime) return result;
+      const startTime = moment(downtime.startTime);
+      const endTime = moment(downtime.endTime);
+      return {
+        startTime: result.startTime && result.startTime.isBefore(startTime) ? result.startTime : startTime,
+        endTime: result.endTime && result.endTime.isAfter(endTime) ? result.endTime : endTime,
+      };
+    }, {});
+  }
+
+  // Converts the array of dependencies/service names into key/value pairs, with service statuses as keys and a list of
+  // downtime information (each downtime corresponding to a dependency/service name) as the values.
+  calculateDowntime(dependencies, scheduledDowntime) {
+    const now = moment();
+    const nextHour = moment().add(1, 'hour');
+    const downtimeMap = {};
+
+    // Create a map with the service statuses as keys { ok, down, downApproaching }
+    Object.keys(serviceStatus).forEach((status) => { downtimeMap[status] = []; });
+
+    return dependencies
+
+      // Get the corresponding downtime value by the name of the service/dependency
+      .map((serviceName) => scheduledDowntime.find(downtime => downtime.service === serviceName))
+
+      // Remove null values (services that have no known downtime)
+      .filter(downtime => !!downtime)
+
+      // Put each value into the corresponding status of the downtimeMap
+      .reduce((map, downtime) => {
+        const status = this.getStatusForDowntime(downtime, now, nextHour);
+        return {
+          ...map,
+          [status]: map[status].concat(downtime)
+        };
+      }, downtimeMap);
+  }
+
+  determineStatus(downtimeMap) {
+    if (this.props.determineStatus) return this.props.determineStatus(downtimeMap);
+
+    const statusDown = downtimeMap[serviceStatus.down].length > 0;
+    const statusDownApproaching = downtimeMap[serviceStatus.downtimeApproaching].length > 0;
+    const statusOk = !statusDown && !statusDownApproaching;
+
+    if (statusOk) return serviceStatus.ok;
+    if (statusDown) return serviceStatus.down;
+
+    return serviceStatus.downtimeApproaching;
+  }
+
+  renderStatusDown({ endTime }) {
+    const title = `The ${this.props.appTitle} is down for maintenance.`;
+    let message = <p>We’re making some updates to {this.props.appTitle}. We’re sorry it’s not working right now. Please check back soon.</p>;
+    if (endTime) {
+      message = (
+        <p>We’re making some updates to {this.props.appTitle}. We’re sorry it’s not working right now, and we hope to be finished by {endTime.format('MMMM Mo, LT')}. Please check back soon.</p>
```

Comment:
Ooh, didn't know `LT` was a thing. Very cool.
Just to make sure all our bases are covered, have you made sure this plays nice with time zones? I know Mark ran into some funny stuff a few days ago with time zones.

Reply:
I believe it should be adjusted for the [locale](https://momentjs.com/docs/#/displaying/)

Reply:
There's also a particular way that the content team has defined for showing AM and PM, that we've had to set directly: https://github.com/department-of-veterans-affairs/vets-website/blob/933acc5b11ab52db6269fcc0ae275cb6b33a06b7/src/js/common/schemaform/save-in-progress/RoutedSavableApp.jsx#L31. So may want that in here, too.

Reply:
Good suggestion, thanks!